hi-ml-histopathology/src/histopathology/scripts/generate_crossval_report.py

```diff
@@ -115,12 +197,46 @@ def render_roc_and_pr_curves(report: HTMLReport, heading: str, level: int,
     report.add_images([roc_pr_curves_fig_path], base64_encode=True)
 
 
+def render_confusion_matrices(report: HTMLReport, heading: str, level: int, class_names: List[str],
+                              parent_run_id: str, aml_workspace: Workspace, report_dir: Path,
+                              output_filename: str, overwrite: bool, prefix: str = '') -> None:
+    """
+    Function to render confusion matrices for HTML reports.
+
+    :param report: HTML report to perform the rendering.
+    :param heading: Heading of the section.
+    :param level: Level of the section (e.g. sub-section, sub-sub-section, and so on).
+    :param class_names: Names of classes.
+    :param parent_run_id: The parent Hyperdrive run ID.
+    :param aml_workspace: Azure ML workspace of the run.
+    :param report_dir: Local directory where the report is stored.
+    :param output_filename: Name of the output CSV file.
+    :param overwrite: Forces (re)download of metrics and output files, even if they already exist locally.
+    :param prefix: Prefix to add to the figures saved (e.g. `val`, `test`).
+    """
+    report.add_heading(heading, level=level)
+    outputs_dfs = collect_crossval_outputs(parent_run_id, report_dir, aml_workspace, output_filename=output_filename,
```

Comment:
I'd suggest calling `collect_crossval_outputs()` only once in the body of `generate_html_report()`, and refactor `render_confusion_matrices()` and `render_roc_and_pr_curves()` to take `outputs_dfs` as an argument instead, to avoid downloading the (fairly large) CSVs multiple times.

Reply:
Great suggestion. I have changed it now